src/twisted/test/test_application.py

```diff
@@ -889,3 +896,54 @@ def install():
                                ['--reactor', 'fakereactortest', 'subcommand'])
         self.assertIn(message, e.args[0])
         self.assertIn("help-reactors", e.args[0])
+
+
+
+class HelpReactorsTests(unittest.TestCase):
+    """
+    --help-reactors lists the available reactors
+    """
+    def setUp(self):
```

Comment:
The reactor list is fairly stable and the platforms on which reactors work almost never change so it's probably pretty reasonable to test this behavior against the real reactor plugins.

There's more than one unit under tests here, though.  First, there's the plugins themselves: epollreactor, iocpreactor, etc.  Then there's the dropin which defines a collection of plugins.  Finally, there's the filtering and reporting logic of `ReactorSelectionMixin`.  The ideal I would strive for, in general, is to separate these units out and test them independently (and then perhaps perform an integration test to ensure they fit together).  In this case, that would suggest parameterizing the `ReactorSelectionMixin` and providing a set of reactor plugins defined by the test suite to it.  The tests is then independent of platform or Python version or changes to the reactors Twisted actually provides (or even third-party reactor plugins which the tests will currently examine - allowing something pathelogical like a Linux port of the iocp reactor - which would break the last test in this TestCase if installed).

I'd consider this optional based on the caveats given at the top.  It might make a nice follow-up refactoring, though.